src/main/scala/firrtl/passes/Passes.scala

```diff
@@ -242,11 +241,12 @@ object Legalize extends Pass {
     }
   }
   def run (c: Circuit): Circuit = {
-    def legalizeE (e: Expression): Expression = {
-      e map (legalizeE) match {
-        case e: DoPrim => legalizeShiftRight(e)
-        case e => e
+    def legalizeE(expr: Expression): Expression = expr map legalizeE match {
+      case prim: DoPrim => prim.op match {
```

Comment:
I didn't unapply the DoPrim because it gets passed to legalizeShiftRight. I could make the legalize functions (more come in a later commit) accept the args, consts, and tpe from the DoPrim but that seems a bit excessive.